packages/flutter_tools/lib/src/doctor_validator.dart

```diff
@@ -58,7 +60,19 @@ abstract class DoctorValidator {
   /// Duration before the spinner should display [slowWarning].
   Duration get slowWarningDuration => _slowWarningDuration;
 
-  Future<ValidationResult> validate();
+  /// Performs validation by invoking [validateImpl].
+  ///
+  /// Tracks time taken to execute the validation step.
+  Future<ValidationResult> validate() async {
+    final Stopwatch stopwatch = Stopwatch()..start();
+    final ValidationResult result = await validateImpl();
+    stopwatch.stop();
```

Comment:
Will we ever reach this if the validator was running when the build got timed out?

Reply:
No, but we do already output a duration to the console if a validation step exceeds the timeout duration.

Reply:
> No, but we do already output a duration to the console if a validation step exceeds the timeout duration.

Which timeout? For example, this change wouldn't help with this situation, right? https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8733230104891163105/+/u/run_build_android_host_app_with_module_source/stdout

Reply:
This timeout: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/doctor.dart#L246

It isn't a solution for the situation we're seeing but it might give us some insights if we have multiple long running doctor validators that push us closer to the CI timeout.

Reply:
Genuine question: Would this PR change anything in the case that a validator completely stalls? Do validators simply get interrupted after running for 00:04:30?

Reply:
No, I don't think it would since the `Future.any([validatorResult, timeout])` would complete after 00:04:30, leaving the result of the validator to be ignored.